src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -55,16 +80,71 @@ Task IOpenInIDERequestHandler.ShowHotspotAsync(IShowHotspotRequest request)
                 throw new ArgumentNullException(nameof(request));
             }
 
+            return ShowHotspotAsync(request);
+        }
+
+        private async Task ShowHotspotAsync(IShowHotspotRequest request)
+        {
+            logger.WriteLine(OpenInIDEResources.OpenHotspot_ProcessingRequest, request.ServerUrl, request.ProjectKey,
+                request.OrganizationKey ?? OpenInIDEResources.OpenHotspot_NullOrganization, request.HotspotKey);
+
+            // Always show the Hotspots tool window. If we can't successfully process the
+            // request we'll show a gold bar in the window
+            toolWindowService.Show(HotspotsToolWindow.ToolWindowId);
+            failureInfoBar.Clear();
+
             if (!ideStateValidator.CanHandleOpenInIDERequest(request.ServerUrl, request.ProjectKey, request.OrganizationKey))
             {
-                return Task.CompletedTask;
+                // We're assuming the validator will have output an explanantion of why the IDE
+                // isn't in the correct state
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspot = await TryGetHotspotData(request.HotspotKey);
+            if (hotspot == null)
+            {
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspotViz = converter.Convert(hotspot);
+            if (hotspotViz == null)
+            {
+                logger.WriteLine(OpenInIDEResources.OpenHotspot_UnableToConvertHotspotData);
+                ShowFailureInfoBar();
+                return;
+            }
+
+            if (!navigator.TryNavigate(hotspotViz))
+            {
+                logger.WriteLine(OpenInIDEResources.ApiHandler_FailedToNavigateToHotspot, hotspotViz.FilePath, hotspotViz.StartLine);
+                ShowFailureInfoBar();
             }
 
-            // TODO
-            // * fetch hotspot data
-            // * push to the data source
-            // * set the selection
-            return Task.CompletedTask;
+            // Add to store regardless of whether navigation succeeded
+            hotspotsStore.Add(hotspotViz);
         }
+
+
+        private void ShowFailureInfoBar()
+        {
+            RunOnUIThread.Run(() => failureInfoBar.Show());
```

Comment:
Trying to show the gold bar on a background thread causes an exception (makes sense, because VS is trying to create WPF elements).
I'll add "CheckAccess" calls to the info bar manager in a separate PR.
Going forwards, we should consider making any new service we write that requires a UI thread/calls a VS service that does async.